src/lib.rs

```diff
@@ -227,31 +231,30 @@ impl Texture {
 }
 
 /// Configuration for the renderer.
-pub struct RendererConfig<'vs, 'fs> {
+pub struct RendererConfig<'s> {
     pub texture_format: TextureFormat,
     pub depth_format: Option<TextureFormat>,
     pub sample_count: u32,
-    pub vertex_shader: Option<ShaderModuleDescriptor<'vs>>,
-    pub fragment_shader: Option<ShaderModuleDescriptor<'fs>>,
+    pub shader: Option<ShaderModuleDescriptor<'s>>,
+    pub vertex_shader_entry_point: String,
```

Comment:
This should just be a &str, we don't expect this struct to live a long time so adding the lifetimes are okay.

Reply:
The `RenderConfig` is currently copied into the `Renderer` and must live as long. The shader uses an interesting pattern of swapping for `None` and assigning a `'static` lifetime, so I mirrored that pattern for the shader entry points. It does mean the user can pass `None` for the entry points which will crash (as they always could with the shader), but it's at least consistent with the existing API.

Reply:
Alright, I think that's weird, but totally shouldn't block this.